src/components/interaction-designer/BuilderCanvas.vue

```diff
@@ -12,21 +12,54 @@
 </template>
 
 <script lang="ts">
-import { mapGetters } from 'vuex'
-import { Component, Vue } from 'vue-property-decorator'
+import { Component, Vue, Watch } from 'vue-property-decorator'
 import Block from '@/components/interaction-designer/Block.vue'
+import { namespace } from 'vuex-class'
+import {IBlock, IFlow} from '@floip/flow-runner'
+import { find, isEqual, cloneDeep } from 'lodash'
+import { IValidationStatus } from '@/store/validation'
 
-  @Component({
-    components: {
-      Block,
-    },
+const flowVuexNamespace = namespace('flow')
+const validationVuexNamespace = namespace('validation')
 
-    computed: {
-      ...mapGetters('flow', ['activeFlow']),
-    },
-
-  })
+@Component({
+  components: {
+    Block,
+  },
+})
 export default class BuilderCanvas extends Vue {
+  @Watch('activeFlow', { deep: true, immediate: true })
+  async onActiveFlowChanged(newFlow: IFlow) {
+    console.debug('active flow has changed, validating ...', newFlow)
+    await this.validate_flow({ flow: newFlow })
+  }
+
+  @Watch('blocksOnActiveFlowForWatcher', { deep: true, immediate: true })
+  async onBlocksInActiveFlowChanged(newBlocks: IBlock[], oldBlocks: IBlock[]) {
+    if (!!newBlocks && newBlocks.length > 0) {
+      console.debug('blocks inside active flow have changed, validating ...');
+      for (let i = 0;  i < newBlocks.length; i++) {
+        const currentNewBlock = newBlocks[i]
+        const currentOldBlock = find(oldBlocks, { 'uuid': currentNewBlock.uuid })
+
+        if (isEqual(currentNewBlock, currentOldBlock)) {
+          continue; // no changes found
+        }
+
+        await this.validate_block({ block: currentNewBlock });
```

Comment:
I'm seeing a lot of semicolons! Is our standard to be semicolon-less?

Reply:
Thanks, had new IDE settings on Ubuntu (spare machine) 😆 , removing all